### ISSUE TITLE
Fiks Arkiv Client Java - not parsing correctly dates in the format YYYY-MM-DD

### ISSUE THREAD

**Issue by myakimov** [ks-no/fiks-arkiv-client-java#6]
Hello,
While moving the parsing of the Fiks Arkiv model to use 0.1.3 version of the libary (maven central artefact) noticed that the validator provided dates for some cases coming in the format YYYY-MM-DD are not parsed and the value remains empty. 

**Example from "Ny Journalpost - Normalsituasjon 6 - Innkommende journalpost"  from validator :** 

```xml
   <dokumentetsDato>2021-03-20</dokumentetsDato>
   <mottattDato>2021-03-22T00:00:00+01:00</mottattDato>
   <offentlighetsvurdertDato>2021-03-22</offentlighetsvurdertDato>
```
The fields dokumentetsDato and offentlighetsvurdertDato remain null in the object representing the arkivmelding, while the mottattDato seems to be parsed successfully.

Tried adding to the dates the time and the zone and all seems to be working fine.

Possible cause for the problem is the mapping of the dates to the `ZonedDateTime` class : 
https://github.com/ks-no/fiks-arkiv-client-java/blob/87c34a3065210dce61c86bf067edec27aa8bc9e5/fiks-arkiv-api/src/main/xjb/bindings.xjb#L8-L19

Currently this prevents handling all the information from the tests of the validator.

Regards,
Marin

**Comment by audunmeltzer** [comment]
Thank you for detailed description

We have release verseion 0.1.3, with date without offset.

**Comment by myakimov** [comment]
Hello, 

Thanks for looking at the issue. What I can see though is that you changed the type of the printMethod, which is related to the generation of XML data and how the date is represended in the generated XML. The issue we experience is when we receive a message containing dates in this format and trying to parse it. In this scenario, the dates are not parsed correctly - they are null in the resulting object.

Looking a bit deeper is the parseMethod specified, parsing a date in the format `uuuu-MM-dd` or `yyyy-MM-dd` is not possivle with just the `java.time.ZonedDateTime.parse` method and there should be either an JAXB Adapter as specified here: http://massfords.com/Use-Java8-time-instead-of-jaxb-gregorian/, or just another method that is properly parsing the string to the ZonedDateTime object.

Parsing a date in the format and having a ZonedDataTime can be achived with something like this (going through LocalDate) : 

```java
// Define a DateTimeFormatter
DateTimeFormatter dtf = DateTimeFormatter.ofPattern("uuuu-MM-dd");

// Given date string
String strDate = "2020-08-24";

// Parse the given date string to ZonedDateTime
ZonedDateTime zdt = LocalDate.parse(strDate, dtf).atStartOfDay(ZoneOffset.UTC);

System.out.println(zdt);
```
I did try creating such a simple method and adding it to the binding instead of the `java.time.ZonedDateTime.parse`  and it seems to be working fine.

**Comment by audunmeltzer** [comment]
In PR https://github.com/ks-no/fiks-arkiv-client-java/pull/8
I have changed the mapping for xs:date to LocalDate. 
In fiks-arkiv-api/src/test/kotlin/no/ks/fiks/io/arkiv/model/ArkivmeldingTest.kt
I have verified this by unmarshall xml back to model for assertion

I think this will solve issue with parsing message to model as described

**Comment by audunmeltzer** [comment]
Should be resolved in release 0.1.5

**Comment by myakimov** [comment]
Thanks @audunmeltzer, the parsing now works fine!
